pandas/tests/frame/test_reshape.py

```diff
@@ -1273,6 +1273,17 @@ def test_stack_timezone_aware_values():
     tm.assert_series_equal(result, expected)
 
 
+def test_stack_empty_frame():
+    tm.assert_series_equal(
+        DataFrame().stack(), Series(index=MultiIndex([[], []], [[], []]), dtype=object)
```

Comment:
Why do you expect that there should be MultiIndex?
Would it be reasonable to expect ``Series([], dtype=object)``?

Reply:
Note that `Series([1, 2, 3]).unstack()` will throw an error as it does not know what to put on the columns, this is because it has only one level of index. I think the same should apply to an empty Series, so `Series([]).unstack()` would throw the same error and not become an empty DataFrame.

Given this constraint, for stack/unstack round-trip to work,  DataFrame([]).stack() needs to return an empty Series with empty multi index with two levels, one from its original index and one from its empty column.